packages/core/src/lib/markdown-it/highlight/HighlightRuleComponent.ts

```diff
@@ -47,7 +41,11 @@ class HighlightRuleComponent {
       // There are four capturing groups: [full match, line number, start bound, end bound]
       const groups = sliceMatch.slice(1); // discard full match
 
-      let lineNumber = parseInt(groups.shift(), 10);
+      const lineNumberStr = groups.shift();
+      if (lineNumberStr === undefined) {
+        return null;
+      }
+      let lineNumber = parseInt(lineNumberStr, 10);
```

Comment:
```suggestion
      let lineNumber = parseInt(groups.shift() ?? '', 10);
```
Since `parseInt('')` will be NaN!